.bin/esyBuildCommand.js

```diff
@@ -0,0 +1,225 @@
+const path = require('path');
+
+function buildCommand(curDir, env, args) {
+
+  // first group is a shared (built-in) env
+  env = env.slice();
+  let builtIns = env.shift();
+
+  let envMap = env.reduce((envMap, pkg) => {
+    envMap.set(pkg.packageJson.name, pkg);
+    return envMap;
+  }, new Map());
+
+  let rules = env
+    .map(pkg => generateMakeRule(builtIns, envMap, pkg))
+    .filter(rule => rule != null);
+  console.log(rules
+    .map(rule => renderMakeRule(rule))
+    .join('\n\n'));
+}
+
+function generateMakeRule(builtIns, envMap, pkg) {
+
+  function envToEnvList(env) {
+    return env.envVars.map(env => ({
+      name: env.name,
+      value: env.normalizedVal
+    }));
+  }
+
+  if (pkg.packageJson.pjc && pkg.packageJson.pjc.build) {
+    return {
+      name: ` *** Build ${pkg.packageJson.name} ***`,
+      target: pkg.packageJson.name,
+      dependencies: pkg.packageJson.dependencies != null
+        ? Object.keys(pkg.packageJson.dependencies)
+        : [],
+      env: []
+        .concat(envToEnvList(builtIns))
+        .concat(getBuildEnv(envMap, pkg))
+        .concat(envToEnvList(pkg)),
+      command: pkg.packageJson.pjc.build,
+    };
+  } else {
+    // TODO: Returning an empty rule. Is that really what we want here?
+    return {
+      name: ` *** Build ${pkg.packageJson.name} ***`,
+      target: pkg.packageJson.name,
+      dependencies: pkg.packageJson.dependencies != null
+        ? Object.keys(pkg.packageJson.dependencies)
+        : [],
+      command: null,
+    };
+  }
+}
+
+function getPkgEnv(envMap, pkg, current = false) {
+  let prefix = current ? 'cur' : (pkg.packageJson.name + '_')
+  let info = pkg.packageJson;
+  return [
+    {
+      name: `${prefix}_name`,
+      value: info.name,
+    },
+    {
+      name: `${prefix}_version`,
+      value: info.version,
+    },
+    {
+      name: `${prefix}_root`,
+      value: pkg.root,
+    },
+    {
+      name: `${prefix}_depends`,
+      value: info.dependencies != null
+        ? `"${Object.keys(info.dependencies).join(' ')}"`
+        : null,
+    },
+    {
+      name: `${prefix}_depends_install_closed`,
+      value: info.dependencies != null
+      ? `"${collectTransitiveDependencies(envMap, info)
+            .map(dep => `$_install_tree/node_modules/${dep}/lib`)
+            .join(':')}"`
+        : null,
```

Comment:
This is hackish, should generate `findlib.conf` as a part of `pic build` preparations instead and not put a burden on a build tool.